statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,612 @@
+        of ones is constructed
+    weights : array-like or None
+        The weight for each observation. If none, an array
+        of ones is constructed
+    nest : boolean
+        allows user to specify if PSU's with the same
+        PSU number in different strata are treated as distinct PSUs.
+
+    Attributes
+    ----------
+    weights : (n, ) array
+        The weight for each observation
+    nstrat : integer
+        The number of district strata
+    sclust : (n, ) array
+        The relabeled cluster array from 0, 1, ..
+    strat : (n, ) array
+        The related strata array from 0, 1, ...
+    ncs : (self.nstrat, ) array
+        Holds the number of clusters in each stratum
+    sfclust : ndarray
+        The stratum for each cluster
+    nclust : integer
+        The total number of clusters across strata
+    """
+
+    def __init__(self, strata=None, cluster=None, weights=None,
+                 rep_weights=None, fpc=None, se_method='jack', nest=True):
+        # Ensure method for SE is supported
+        if se_method not in ["boot", 'mean_boot', 'jack']:
+            raise ValueError("Method %s not supported" % se_method)
+        else:
+            self.se_method = se_method
+
+        self.rep_weights = rep_weights
+
+        if self.rep_weights is not None:
+            if strata is not None or cluster is not None:
+                raise ValueError("If providing rep_weights, do not provide \
+                             cluster or strata")
+            if weights is None:
+                self.weights = np.ones(self.rep_weights.shape[0])
+            else:
+                self.weights = weights
+            return
+
+        strata, cluster, self.weights, \
+            self.fpc = self._check_args(strata, cluster, weights, fpc)
+
+        # Recode strata and clusters as integer values 0, 1, ...
+        _, self.strat = np.unique(strata, return_inverse=True)
+        _, clust = np.unique(cluster, return_inverse=True)
+
+        # the number of distinct strata
+        self.nstrat = max(self.strat) + 1
+
+        # If requested, recode the PSUs to be sure that the same PSU # in
+        # different strata are treated as distinct PSUs. This is the same
+        # as the nest option in R.
+        if nest:
+            m = max(clust) + 1
+            sclust = clust + m*self.strat
+            _, self.sclust = np.unique(sclust, return_inverse=True)
+        else:
+            self.sclust = clust.copy()
+
+        # The number of clusters per stratum
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.ncs = np.bincount(self.strat[ii])
+
+        # The stratum for each cluster
+        self.sfclust = self.strat[ii]
+
+        # The fpc for each cluster
+        self.fpc = self.fpc[ii]
+
+        # The total number of clusters over all stratum
+        self.nclust = np.sum(self.ncs)
+
+        # get indices of all clusters within a stratum
+        self.ii = []
```

Comment:
But it appears to be indices of strata, not indices of clusters, so maybe `strat_inds`.